File: Python3/1578. Minimum Time to Make Rope Colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        ans = 0
        p, pre = 0, colors[0]
        
        for c, cur in enumerate(colors[1:], 1):
            if pre == cur:
                # release current balloon
                if neededTime[c] < neededTime[p]:
                    ans += neededTime[c]
                    
                # release previous balloon
                else:
                    ans += neededTime[p]
                    p, pre = c, cur
            else:
                p, pre = c, cur
                    
        return ans

File: Java/1578. Minimum Time to Make Rope Colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        int ans = 0, p = 0;
        char pre = colors.charAt(0);
        
        for (int c = 1; c < colors.length(); c++) {
            char cur = colors.charAt(c);
            if (pre == cur) {
                if (neededTime[c] < neededTime[p]) {
                    ans += neededTime[c];
                } else {
                    ans += neededTime[p];
                    p = c;
                }
            } else {
                p = c;
                pre = cur;
            }
        }
        
        return ans;
    }
}